qiskit/pulse/__init__.py

```diff
@@ -130,15 +145,98 @@
 
 """
 
-from .channels import (DriveChannel, MeasureChannel, AcquireChannel,
-                       ControlChannel, RegisterSlot, MemorySlot)
-from .commands import AcquireInstruction, FrameChange, PersistentValue
-from .configuration import LoConfig, LoRange, Kernel, Discriminator
-from .exceptions import PulseError
-from .instruction_schedule_map import InstructionScheduleMap
-from .instructions import Acquire, Instruction, Delay, Play, ShiftPhase, Snapshot, SetFrequency
-from .interfaces import ScheduleComponent
-from .pulse_lib import (SamplePulse, Gaussian, GaussianSquare, Drag,
-                        Constant, ConstantPulse, ParametricPulse)
-from .pulse_lib.samplers.decorators import functional_pulse
-from .schedule import Schedule
+# Builder imports.
+from qiskit.pulse.builder import (
+    # builder construction methods.
+    active_backend,
+    active_transpiler_settings,
+    active_circuit_scheduler_settings,
+    append_instruction,
+    build,
+    num_qubits,
+    qubit_channels,
+    samples_to_time,
+    time_to_samples,
+    # builder instructions.
+    acquire,
+    barrier,
+    call,
+    call_schedule,
+    delay,
+    play,
+    set_frequency,
+    set_phase,
+    shift_frequency,
+    shift_phase,
+    snapshot,
+    # builder channels.
+    acquire_channel,
+    control_channels,
+    drive_channel,
+    measure_channel,
+    # builder contexts.
+    align_left,
+    align_right,
+    align_sequential,
+    circuit_scheduler_settings,
+    frequency_offset,
+    group,
+    inline,
+    pad,
+    phase_offset,
+    transpiler_settings,
+    # builder macros.
+    measure,
+    measure_all,
+    delay_qubits,
+    # builder circuit instructions.
+    call_circuit,
+    call_gate,
+    cx,
+    u1,
+    u2,
+    u3,
+    x
```

Comment:
can we document somewhere in these functions that they will be temporary to enable composing complex schedules, but that they will be removed in favor of a builder interface for building timed circuits. Specifically I think we shouldn't have these 3 U gates hardcoded as they are being replaced (https://github.com/Qiskit/qiskit-terra/issues/4106) but if they are temporary I'm ok.

Reply:
I'm fine adding them temporarily. Realistically I will end up having to do #4106 and consequently, this is a ways away.

Reply:
Added a note on future deprecation in 1ccad5174